### PR TITLE
Implementation for Alerts component.

### PR DESCRIPTION
**Pull request by cardosogabriel**
Writing the alert components in patternfly-atomic style.

### CODE REVIEW

File: source/scss/_pf-basic-variables.scss [hunk 1]
```diff
@@ -456,3 +457,32 @@ $tag-font-weight:           $pf-font-weight-bold;
 // Use a higher than normal value to ensure completely rounded edges when
 // customizing padding or font-size on labels.
 // $tag-pill-border-radius:    10rem !default;
+
+
+// Form states and alerts
+//
+// Define colors for form feedback states and, by default, alerts.
+
+$state-success-text:             $color-pf-black-800 !default;
```
Comment (andresgalante) [inline source/scss/_pf-basic-variables.scss]: I think we need to remove the `!default` in order for our changes to be applyed

File: source/scss/patternfly.scss [hunk 2]
```diff
@@ -27,6 +27,7 @@
 @import "node_modules/bootstrap/scss/list-group";
 @import "node_modules/bootstrap/scss/responsive-embed";
 @import "node_modules/bootstrap/scss/close";
+@import "pf-component-alerts";
```
Comment (andresgalante) [inline source/scss/patternfly.scss]: Please put this line right after the one we import bootstrap alerts

File: source/scss/_pf-basic-icons.scss [hunk 3]
```diff
@@ -112,6 +112,7 @@
   content: $pficon-var-import;
 }
 .#{$pf-icon-prefix}-info:before {
+  color: $color-pf-black-700;
```
Comment (cardosogabriel) [inline source/scss/_pf-basic-icons.scss]: The 4 alert icons have a different color than the normal gray. 
The declaration for this color was in the alerts.less. I moved together with the others in the icons file.

File: source/scss/_pf-basic-variables.scss (hunk 1 above)
Comment (cardosogabriel) [inline source/scss/_pf-basic-variables.scss]: They do work with the !default.
I found this explanation about the usage: https://robots.thoughtbot.com/sass-default
Basically 

> This means that if the variable has already been assigned to, it won’t be re-assigned, but if it doesn’t have a value yet, it will be given one.

So I assume that no variables defined by us should have the default? I see a bunch of others with default in this file...

File: source/scss/_pf-component-alerts.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    color: $link-color;
+    &:hover {
+      color: $link-hover-color;
+    }
+  }
+}
+
+.alert-icon {
+  font-size: $alert-close-font-size;
+  position: absolute;
```
Comment (cardosogabriel) [inline source/scss/_pf-component-alerts.scss]: Updated to use flexbox.
Needed to add a new <div> around the text, but removed the positions ;)

File: source/_patterns/components/00-alerts/00-alert.hbs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div class="alert {{ alertClass }}" role="alert">
+  <span class="alert-icon pficon {{ alertIcon }}"></span>
+  <div class="alert-text">
```
Comment (cardosogabriel) [inline source/_patterns/components/00-alerts/00-alert.hbs]: @andresgalante , should this class be "alert__text", or even "alert-pf__text"?

Comment (andresgalante) [inline source/_patterns/components/00-alerts/00-alert.hbs]: Yes, if it's a new class name we are creating, then it should be BEM and prefixed with `pf-`

File: source/scss/_pf-component-alert.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+// Alerts
+// --------------------------------------------------
+
+.alert {
+  padding-left: $spacer;
+  padding-right: $spacer;
+  display: flex;
+  align-items: center;
+
+  .alert-link {
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: Can we avoid nesting `alert-link` here?

File: source/scss/_pf-component-alert.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.alert-dismissible .close {
+  order: 1;
+  right: 0;
+  top: 0;
+  margin-left: $spacer;
+  opacity: .85;
+
+  @include hover-focus {
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: 😄 nice! always use hover and hover-focus with the mixin

File: source/scss/_pf-component-alert.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
+//
+// Alerts
+// --------------------------------------------------
+
+.alert {
+  padding-left: $spacer;
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: Can you change the variables `$alert-padding-x` and `$alert-padding-y` instead of doing it here?

File: source/scss/_pf-component-alert.scss [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  padding-right: $spacer;
+  display: flex;
+  align-items: center;
+
+  .alert-link {
+    color: $link-color;
+    &:hover {
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: please use the mixin hover

File: source/scss/_pf-component-alert.scss [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+.alert-text {
+  flex-grow: 1;
+}
+
+.alert-dismissible .close {
+  order: 1;
+  right: 0;
+  top: 0;
+  margin-left: $spacer;
+  opacity: .85;
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: Can we create `pf-` variables for opacity so we reuse it on other components?

File: source/scss/_pf-component-alert.scss [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+.alert-icon {
+  font-size: $alert-close-font-size;
+  margin-right: $spacer;
+}
+
+.alert-text {
+  flex-grow: 1;
+}
+
+.alert-dismissible .close {
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: I know we will have to do the nesting to position the "x".
The close component will be reuse by other components. Can we create the close element as a component and the include it in other places? Define it once with as much patternfly styles as we can and then add here only the  styles needed for alerts.
also, can we nest `.close` so it's more clear?

File: source/scss/_pf-component-alert.scss [hunk 12]
```diff
@@ -0,0 +1,38 @@
+//
+// Alerts
+// --------------------------------------------------
+
+.alert {
+  padding-left: $spacer;
+  padding-right: $spacer;
+  display: flex;
+  align-items: center;
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: I'd say this should be `flex-start` since we want them align to the top when the text wrap into more than one line.

**Comment by andresgalante** [comment]
@cardosogabriel sorry for SO many reviews. I was really picky.

File: source/scss/_pf-component-alert.scss (hunk 6 above)
Comment (cardosogabriel) [inline source/scss/_pf-component-alert.scss]: It comes nested from patternfly. So I could unnest if I create a specific class. Do u think it worth it?

File: source/scss/_pf-component-alert.scss (hunk 8 above)
Comment (cardosogabriel) [inline source/scss/_pf-component-alert.scss]: Nice catch, I was unaware of such variables :)

File: source/scss/patternfly.scss (hunk 2 above)
Comment (cardosogabriel) [inline source/scss/patternfly.scss]: Done.

File: source/_patterns/components/00-alerts/00-alert.hbs (hunk 5 above)
Comment (cardosogabriel) [inline source/_patterns/components/00-alerts/00-alert.hbs]: Done, thanks.

File: source/scss/_pf-component-alert.scss (hunk 9 above)
Comment (cardosogabriel) [inline source/scss/_pf-component-alert.scss]: Done.

File: source/scss/_pf-component-alert.scss (hunk 10 above)
Comment (cardosogabriel) [inline source/scss/_pf-component-alert.scss]: Done.

File: source/scss/_pf-component-alert.scss (hunk 12 above)
Comment (cardosogabriel) [inline source/scss/_pf-component-alert.scss]: Done, removed center, which makes flex-start (default).

**Comment by cardosogabriel** [comment]
@andresgalante Thanks for such detailed review.

File: source/scss/_pf-component-alert.scss [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  font-size: $alert-close-font-size;
+  margin-right: $spacer;
+}
+
+.pf-alert__text {
+  flex-grow: 1;
+}
+
+.alert-dismissible {
+  .close {
```
Comment (andresgalante) [inline source/scss/_pf-component-alert.scss]: I don't get this, I'll ping you when you are around and let's talk about it.

Comment (cardosogabriel) [inline source/scss/_pf-component-alert.scss]: Removed the right: -1rem declaration that already comes from bootstrap.

**Comment by cardosogabriel** [comment]
Rebased the PR.

File: source/scss/_pf-basic-variables.scss [hunk 14]
```diff
@@ -456,3 +460,33 @@ $tag-font-weight:           $pf-font-weight-bold;
 // Use a higher than normal value to ensure completely rounded edges when
 // customizing padding or font-size on labels.
 // $tag-pill-border-radius:    10rem !default;
+
+
+// Form states and alerts
+//
+// Define colors for form feedback states and, by default, alerts.
```
Comment (jgiardino) [inline source/scss/_pf-basic-variables.scss]: Should icon colors also be included here so that all the colors are defined in one place?

File: source/scss/_pf-basic-variables.scss [hunk 15]
```diff
@@ -456,3 +460,33 @@ $tag-font-weight:           $pf-font-weight-bold;
...
+$state-danger-border:            $brand-danger;
+
+
+// Alerts
+//
+// Define alert colors, border radius, and padding.
+
+$alert-padding-x:             $spacer;
+$alert-link-font-weight:      normal;
+$alert-close-font-size:       1.5rem;
```
Comment (jgiardino) [inline source/scss/_pf-basic-variables.scss]: Do we currently have varying sizes of the close icon? Should we just have one size for all close icons? E.g. $close-font-size?

Comment (srambach) [inline source/scss/_pf-basic-variables.scss]: Can this use one of the previously defined font sizes like $font-size-relative-lg or $font-size-h3? Neither is 1.5rem but it would be good to stick to the set points we have or add another in the set if needed.

File: source/scss/_pf-basic-variables.scss (hunk 14 above)
Comment (cardosogabriel) [inline source/scss/_pf-basic-variables.scss]: Are you proposing to move the color specs of the icons error, warning, info and success from the _pf-basic-icons.scss to this file?
I'm not strong about this move but I think it works also. But not under this "alerts" section of code, since the icons are used also in other components, like cards. Anyways I'd leave this for a different PR.

File: source/scss/_pf-basic-variables.scss (hunk 15 above)
Comment (cardosogabriel) [inline source/scss/_pf-basic-variables.scss]: Actually we do. The close icon in current toolbar is 14px, and in modals / alerts / toast notifications it is 18px.
It seems that 18px would be too big for the toolbar, and 14px too small for the other ones.

Comment (cardosogabriel) [inline source/scss/_pf-basic-variables.scss]: Actually I was using this variable for the icon size in alerts. So I updated the variable name to $pf-alert-icon-font-size:     $icon-size-sm; 
Introduced the icon-size-sm variable, since we have other icon-size- and none of the font-sizes fit.
